packages/help-center/src/components/index.d.ts

```diff
@@ -147,3 +147,17 @@ declare module '@automattic/state-utils' {
 }
 
 declare module 'calypso/lib/mobile-app';
+
+declare module 'calypso/lib/explat' {
```

Comment:
Hmm, I'm not sure why these types are necessary, the explat client library is fully typed.

Reply:
Yeah, I'm not sure either; just a shot in the dark since I was getting unit-tests failing without them.

Reply:
This is what I was getting [#](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6843_449.6841&logFilter=debug&logView=flowAware):
```
....STDOUT....

[09:28:10 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6844_6844&logFilter=debug&logView=flowAware)
    client/lib/explat/internals/log-error.ts(1,19): error TS7016: Could not find a declaration file for module 'calypso/lib/wp'. '/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/client/lib/wp/node.js' implicitly has an 'any' type.

[09:28:10 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6845_6845&logFilter=debug&logView=flowAware)
      Try `npm i --save-dev @types/wp` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/lib/wp';`

[09:28:10 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6846_6846&logFilter=debug&logView=flowAware)
    client/lib/explat/internals/log-error.ts(2,27): error TS7016: Could not find a declaration file for module 'calypso/server/lib/logger'. '/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/client/server/lib/logger/index.js' implicitly has an 'any' type.

[09:28:10 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6847_6847&logFilter=debug&logView=flowAware)
      Try `npm i --save-dev @types/calypso` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/server/lib/logger';`

[09:28:10 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6848_6848&logFilter=debug&logView=flowAware)
    client/lib/explat/internals/fetch-experiment-assignment.ts(1,19): error TS7016: Could not find a declaration file for module 'calypso/lib/wp'. '/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/client/lib/wp/node.js' implicitly has an 'any' type.

[09:28:10 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/10013369?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=10013369_6849_6849&logFilter=debug&logView=flowAware)
      Try `npm i --save-dev @types/wp` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/lib/wp';`
      ```

Reply:
Ok, I don't know what I'm doing here :) @alshakero can you help me understand what I need to do here?

<details><summary>These are the errors I get if I run `npm run build` in the help-center package directory:</summary>
<p>

```
../../client/lib/explat/internals/log-error.ts:1:19 - error TS7016: Could not find a declaration file for module 'calypso/lib/wp'. '/Users/yoavf/dev/wp-calypso/client/lib/wp/node.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/wp` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/lib/wp';`

1 import wpcom from 'calypso/lib/wp';
                    ~~~~~~~~~~~~~~~~

../../client/lib/explat/internals/log-error.ts:2:27 - error TS7016: Could not find a declaration file for module 'calypso/server/lib/logger'. '/Users/yoavf/dev/wp-calypso/client/server/lib/logger/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/calypso` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/server/lib/logger';`

2 import { getLogger } from 'calypso/server/lib/logger';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../client/lib/explat/internals/fetch-experiment-assignment.ts:1:19 - error TS7016: Could not find a declaration file for module 'calypso/lib/wp'. '/Users/yoavf/dev/wp-calypso/client/lib/wp/node.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/wp` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/lib/wp';`

1 import wpcom from 'calypso/lib/wp';
                    ~~~~~~~~~~~~~~~~

../../client/lib/explat/internals/log-error.ts:1:19 - error TS7016: Could not find a declaration file for module 'calypso/lib/wp'. '/Users/yoavf/dev/wp-calypso/client/lib/wp/node.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/wp` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/lib/wp';`

1 import wpcom from 'calypso/lib/wp';
                    ~~~~~~~~~~~~~~~~

../../client/lib/explat/internals/log-error.ts:2:27 - error TS7016: Could not find a declaration file for module 'calypso/server/lib/logger'. '/Users/yoavf/dev/wp-calypso/client/server/lib/logger/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/calypso` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/server/lib/logger';`

2 import { getLogger } from 'calypso/server/lib/logger';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../client/lib/explat/internals/fetch-experiment-assignment.ts:1:19 - error TS7016: Could not find a declaration file for module 'calypso/lib/wp'. '/Users/yoavf/dev/wp-calypso/client/lib/wp/node.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/wp` if it exists or add a new declaration (.d.ts) file containing `declare module 'calypso/lib/wp';`

1 import wpcom from 'calypso/lib/wp';
                    ~~~~~~~~~~~~~~~~
```


</p>
</details>

Reply:
I suppose the lib itself is typed, but its instance in Calypso depends on untyped dependencies.

Namely this: https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/explat/internals/fetch-experiment-assignment.ts#L1